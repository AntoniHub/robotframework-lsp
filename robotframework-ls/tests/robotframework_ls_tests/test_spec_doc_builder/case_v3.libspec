<?xml version="1.0" encoding="UTF-8"?>
<keywordspec name="case_argspec" type="LIBRARY" format="HTML" scope="GLOBAL" generated="2020-10-26T12:43:31Z" specversion="3" source="case_argspec.py" lineno="1">
<version></version>
<doc>&lt;p&gt;Documentation for library &lt;code&gt;case_argspec&lt;/code&gt;.&lt;/p&gt;</doc>
<inits>
</inits>
<keywords>
<kw name="Arg String Default" lineno="42">
<arguments repr="arg1=my:,my:something,*foo,**foo,&lt;rara&gt;">
<arg kind="POSITIONAL_OR_NAMED" required="false" repr="arg1=my:,my:something,*foo,**foo,&lt;rara&gt;">
<name>arg1</name>
<default>my:,my:something,*foo,**foo,&lt;rara&gt;</default>
</arg>
</arguments>
<doc></doc>
</kw>
<kw name="Arg With Enum Type" lineno="26">
<arguments repr="model: SomeEnum { v1 | v2 }">
<arg kind="POSITIONAL_OR_NAMED" required="true" repr="model: SomeEnum { v1 | v2 }">
<name>model</name>
<type>SomeEnum { v1 | v2 }</type>
</arg>
</arguments>
<doc></doc>
</kw>
<kw name="Arg With Enum Type And Default" lineno="30">
<arguments repr="model: SomeEnum { v1 | v2 } = v2">
<arg kind="POSITIONAL_OR_NAMED" required="false" repr="model: SomeEnum { v1 | v2 } = v2">
<name>model</name>
<type>SomeEnum { v1 | v2 }</type>
<default>v2</default>
</arg>
</arguments>
<doc></doc>
</kw>
<kw name="Arg With Positional" lineno="38">
<arguments repr="arg1, *, arg2">
<arg kind="POSITIONAL_OR_NAMED" required="true" repr="arg1">
<name>arg1</name>
</arg>
<arg kind="NAMED_ONLY_MARKER" required="false" repr="*">
</arg>
<arg kind="NAMED_ONLY" required="true" repr="arg2">
<name>arg2</name>
</arg>
</arguments>
<doc></doc>
</kw>
<kw name="Arg With Starargs" lineno="34">
<arguments repr="arg1, arg2=10, *arg3, **arg4">
<arg kind="POSITIONAL_OR_NAMED" required="true" repr="arg1">
<name>arg1</name>
</arg>
<arg kind="POSITIONAL_OR_NAMED" required="false" repr="arg2=10">
<name>arg2</name>
<default>10</default>
</arg>
<arg kind="VAR_POSITIONAL" required="false" repr="*arg3">
<name>arg3</name>
</arg>
<arg kind="VAR_NAMED" required="false" repr="**arg4">
<name>arg4</name>
</arg>
</arguments>
<doc></doc>
</kw>
<kw name="Arg With Type" lineno="20">
<arguments repr="model: int">
<arg kind="POSITIONAL_OR_NAMED" required="true" repr="model: int">
<name>model</name>
<type>int</type>
</arg>
</arguments>
<doc>&lt;p&gt;Some doc in arg_with_type.&lt;/p&gt;</doc>
</kw>
<kw name="Simple Arg" lineno="14">
<arguments repr="model">
<arg kind="POSITIONAL_OR_NAMED" required="true" repr="model">
<name>model</name>
</arg>
</arguments>
<doc>&lt;p&gt;Some doc in simple_arg.&lt;/p&gt;</doc>
</kw>
</keywords>
</keywordspec>
