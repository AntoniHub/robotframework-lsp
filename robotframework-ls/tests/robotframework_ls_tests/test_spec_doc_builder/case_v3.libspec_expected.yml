Arg String Default:
  args:
  - arg_type: null
    default_value: my:,my:something,*foo,**foo,<rara>
    is_keyword: false
    is_star: false
    name: arg1
    original_arg: arg1=my:,my:something,*foo,**foo,<rara>
Arg With Enum Type:
  args:
  - arg_type: SomeEnum { v1 | v2 }
    default_value: null
    is_keyword: false
    is_star: false
    name: model
    original_arg: 'model: SomeEnum { v1 | v2 }'
Arg With Enum Type And Default:
  args:
  - arg_type: SomeEnum { v1 | v2 }
    default_value: v2
    is_keyword: false
    is_star: false
    name: model
    original_arg: 'model: SomeEnum { v1 | v2 } = v2'
Arg With Positional:
  args:
  - arg_type: null
    default_value: null
    is_keyword: false
    is_star: false
    name: arg1
    original_arg: arg1
  - arg_type: null
    default_value: null
    is_keyword: false
    is_star: false
    name: arg2
    original_arg: arg2
Arg With Starargs:
  args:
  - arg_type: null
    default_value: null
    is_keyword: false
    is_star: false
    name: arg1
    original_arg: arg1
  - arg_type: null
    default_value: '10'
    is_keyword: false
    is_star: false
    name: arg2
    original_arg: arg2=10
  - arg_type: null
    default_value: null
    is_keyword: false
    is_star: true
    name: '@arg3'
    original_arg: '*arg3'
  - arg_type: null
    default_value: null
    is_keyword: true
    is_star: false
    name: '&arg4'
    original_arg: '**arg4'
Arg With Type:
  args:
  - arg_type: int
    default_value: null
    is_keyword: false
    is_star: false
    name: model
    original_arg: 'model: int'
Simple Arg:
  args:
  - arg_type: null
    default_value: null
    is_keyword: false
    is_star: false
    name: model
    original_arg: model
